enumerable_project.rb

```diff
@@ -0,0 +1,139 @@
+ Copyright (c) Muhammasd Arslan
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    return self unless block_given?
+
+    (0..length - 1).each do |i|
+      yield(self[i])
+    end
+    self
+  end
+
+  def my_each_with_index
+    return self unless block_given?
+
+    (0..length - 1).each do |i|
+      yield(self[i], i)
+    end
+  end
+
+  def my_select
+    return self unless block_given?
+
+    arr = []
+    my_each do |x|
+      arr << x if yield(x) == true
+    end
+    arr
+  end
```

Comment:
my_select

Shuld return an Enumerator if no block is given:
`p [1, 2, 3].my_select` returns `#<Enumerator>`